Packs/FeedTAXII/Integrations/FeedTAXII2/FeedTAXII2.py

```diff
@@ -0,0 +1,188 @@
+    )
+    if indicators:
+        return CommandResults(
+            outputs_prefix=CONTEXT_PREFIX + ".Indicators",
+            outputs_key_field="value",
+            outputs=indicators,
+            readable_output=md,
+        )
+    return md
+
+
+def get_collections_command(client):
+    """
+    Get the available collections in the TAXII server
+    :param client: FeedClient
+    :return: available collections
+    """
+    collections = list()
+    for collection in client.collections:
+        collections.append({"Name": collection.title, "ID": collection.id})
+    md = tableToMarkdown("TAXII2 Server Collections:", collections, ["Name", "ID"])
+    return CommandResults(
+        outputs_prefix=CONTEXT_PREFIX + ".Collections",
+        outputs_key_field="ID",
+        outputs=collections,
+        readable_output=md,
+    )
+
+
+def reset_fetch_command(client):
+    """
+    Reset the last fetch from the integration context
+    """
+    demisto.setIntegrationContext({})
+    return (
+        "Fetch was reset successfully. Your next indicator fetch will collect indicators from "
+        'the configured "First Fetch Time"'
+    )
+
+
+def main():
+    params = demisto.params()
+    args = demisto.args()
+    url = params.get("url")
+    collection_to_fetch = params.get("collection_to_fetch")
+    credentials = params.get("credentials") or {}
+    username = credentials.get("identifier")
+    password = credentials.get("password")
+    proxies = handle_proxy()
+    verify_certificate = not params.get("insecure", False)
+
+    command = demisto.command()
+    demisto.info(f"Command being called is {command}")
+
+    try:
+        client = Taxii2FeedClient(
+            url, collection_to_fetch, proxies, verify_certificate, username, password
+        )
+        client.initialise()
+        commands = {
+            "taxii2-reset-fetch-indicators": reset_fetch_command,
+            "taxii2-get-indicators": get_indicators_command,
+            "taxii2-get-collections": get_collections_command,
+        }
+
+        if demisto.command() == "test-module":
+            # This is the call made when pressing the integration Test button.
+            test_module(client)
+
+        elif demisto.command() == "fetch-indicators":
+            initial_interval = params.get("initial_interval")
+            limit = try_parse_integer(params.get("limit") or -1)
+            now = datetime.now()  # might refetch some indicators the next fetch
+            integration_ctx = demisto.getIntegrationContext() or {}
+            last_run = integration_ctx.get(INTEGRATION_CONTEXT_TIME_KEY)
+            indicators = fetch_indicators_command(
+                client, initial_interval, limit, last_run
+            )
+            for iter_ in batch(indicators, batch_size=2000):
+                demisto.createIndicators(iter_)
+
+            demisto.setIntegrationContext(
+                {INTEGRATION_CONTEXT_TIME_KEY: now.timestamp()}
+            )
```

Comment:
aren't we fetching all and doing `suddendeath` as the expirationpolicy?
why fetch only the difference? performance?

Reply:
https://github.com/demisto/etc/issues/22586#issuecomment-643747023